manifests/ubuntu/agent6.pp

```diff
@@ -33,8 +36,8 @@
     location => $location,
     release  => $release,
     repos    => $repos,
-    require  => Package['apt-transport-https'],
-    notify   =>  Exec['apt_update'],
+    key      => $key,
+    require  => Exec['apt-transport-https'],
```

Comment:
If we decide to keep `ensure_packages`, we could probably remove this line as `ensure_packages` would do its thing and ensure the package is available. I wonder if the reason this was here was due to the order in which puppet applied these statements, we'll have to test that out.